base/applications/osk/main.c

```diff
@@ -428,6 +429,60 @@ BOOL OSK_ReleaseKey(WORD ScanCode)
     return TRUE;
 }
 
+/***********************************************************************
+ *
+ *           OSK_ThemeHandler
+ *
+ *  Function helper which handles theme drawing of controls
+ */
+LRESULT APIENTRY OSK_ThemeHandler(HWND hDlg, NMCUSTOMDRAW *pNmDraw)
+{
+    INT iState = PBS_NORMAL;
+
+    /* Check if the application is themed or has a theme active for it */
+    if (IsAppThemed() && IsThemeActive())
+    {
+        Globals.hTheme = OpenThemeData(hDlg, L"Button");
```

Comment:
Do we need to open this every call of `OSK_ThemeHandler` ?
That seems a bit overkill...

Reply:
MSDN specifically says that you've to call `OpenThemeData()` to begin with the applying of theme styles to the window. The function has to be called each time the theme has changed in order to retrieve the handle of the new theme (see `WM_THEMECHANGED`).

Reply:
Why not doing it once on the initialization?

Since you're opening it again and again on WM_NOTIFY (and by the way not closing it), the WM_THEMECHANGED does not have any effect (but it supposed to be).

Reply:
You should check how calc.exe deals with that.

Reply:
This is indeed an overkill. There are two correct ways to do that. 

The first one is to call OpenThemeData when WM_THEMECHANGED is received and call GetWindowTheme during a painting operation. 

The second way is to call OpenThemeData when WM_THEMECHANGED is received, cache its value and use it during a painting operation.